### ISSUE TITLE
Not compatible with Catalina 

### ISSUE THREAD

**Issue by jtcristina** [tSoniq/alphasync#4]
64-bit architecture required and all that. I don't know how difficult it'd be to update given that, but I figured I'd alert you to it as I couldn't get it to work in Catalina 10.15 beta for this reason.

**Comment by gingerbeardman** [comment]
Did you try building from source for 64-bit in Xcode?

If it didn't work where did it error?

**Comment by ohnoai** [comment]
Could you perhaps direct me to some documentation on how to do this? I've googled. I am also super new at coding stuff, like weeks old. @gingerbeardman

**Comment by gingerbeardman** [comment]
All I can do is give you steps that work for me:

### Prep:
1. Download and run [Xcode](https://apps.apple.com/gb/app/xcode/id497799835?mt=12)
2. Download [this project source code](https://github.com/tSoniq/alphasync/archive/master.zip)
3. Unzip `alphasync-master.zip` and open the file `alphasync-master/AlphaSync/AlphaSync.xcodeproj`

You'll get this warning:
![Screen shot 2019-10-13 at 13 14 53](https://user-images.githubusercontent.com/49612/66715561-b488a780-edbc-11e9-9aea-2bf3c607cb33.png)


### Building:
1. In Xcode click the ▶️ "play" button
2. Solve the various build errors (this step is too much to explain here)
3. Repeat until build is successful

### Results:

I did this (on Mojave) and got this build: [AlphaSync-64bit-build.zip](https://github.com/tSoniq/alphasync/files/3721714/AlphaSync-64bit-build.zip)

### Caveat:

If that doesn't work for you in any way, I'm sorry I cannot help any further.

I have no intention of upgrading to Catalina, so you're on your own with this.

**Comment by ohnoai** [comment]
Thank you! You’re a good man.

**Comment by gingerbeardman** [comment]
So it worked for you?

**Comment by ohnoai** [comment]
Technically it did not, I will spend more time with it and pick some minds and I'll update here if I can come up with anything. I appreciate your help though I'm sure it will help others who aren't on Catalina beta.
![Screen Shot 2019-10-13 at 8 36 51 PM](https://user-images.githubusercontent.com/30525022/66728364-3ac2df00-edf9-11e9-86a6-6f8071cb12d8.png)

**Comment by bumper314** [comment]
@joeynw After you get that dialog, go into System Preferences > Security & Privacy > General tab, and you should see a button that allows you to launch AlphaSync anyway.

**Comment by gingerbeardman** [comment]
> Catalina beta

At this point in time Catalina has been released for all on the Mac App Store.

Anyway, the only way to get a build without this warning is for the original developer to issue it by jumping through some Apple hoops.

As @bumper314 says: https://support.apple.com/en-gb/HT202491

![image](https://user-images.githubusercontent.com/49612/66739818-71384280-ee69-11e9-9d11-e3b46749294b.png)

![image](https://user-images.githubusercontent.com/49612/66739883-a349a480-ee69-11e9-93b5-ed1505b3bc3a.png)

**Comment by bumper314** [comment]
@gingerbeardman 

> the only way to get a build without this warning is for the original developer to issue it by jumping through some Apple hoops.

Doesn't need to be the "original developer," anyone with an Apple Developer account ($99/yr) can sign a build, send it to Apple to get [notarized](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution), then release it.  But yes, lots of hoops.

**Comment by ohnoai** [comment]
Thanks. The application appears to be working fine, but the Neo2 is still giving me a "invalid request" error. I have a second device I'll test, but it seems others have been able to use their Neo2 on Catalina with the build above, so I'll close this issue.

**Comment by gingerbeardman** [comment]
Yes, good point @bumper314 — what I meant is that I'm not going to do that in this instance

**Comment by ReinerKnudsen** [comment]
Amazing job, thanks

**Comment by tekemperor** [comment]
@gingerbeardman Would you be willing to create a PR with your changes on this project, or a fork, so others can build it locally?

I am trying to switch the line endings from "Carriage Return" to "Line Feed" to match modern OSX without having to run through an extra script. However, I am stuck in the "Solve the various build errors..." step from your previous comment.

**Comment by gingerbeardman** [comment]
Here we go: https://github.com/tSoniq/alphasync/pull/5

I just had to do this again from scratch because I deleted the previous files.

All except one change are just wrangling Xcode Project Settings.

1. Set ARCHS = "$(ARCHS_STANDARD)" which today means 64-bit, for all Targets
2. Set ALWAYS_SEARCH_USER_PATHS = NO, for all Targets
3. Set MACOSX_DEPLOYMENT_TARGET = 10.9, for all Targets
4. Delete code-signing Build Phase, for both AlphaSync and AlphaSyncLauncher
5. Comment out one unecessary line that was throwing an error, only in AlphaSync

This builds with the current (at time of writing) Xcode 11.3, targeting macOS 10.9 (project was historically set to target 10.5 but that was removed from Xcode a long, long time ago)

Enjoy and Good Luck!
